### ISSUE TITLE
Custom script working like weird

### ISSUE THREAD

**Issue by Neptunespace** [Sonarr/Sonarr#5986]
### Is there an existing issue for this?

- [X] I have searched the existing open and closed issues

### Current Behavior

Hello guy :D
Sorry for my english first :c

Some file that i import trigger my custom script correctly, some don't triggers it.

### Expected Behavior

All the file needs to triggers my custom script.

The problem is here for Ahsoka, all 3 episodes have the same problem (auto import with qbittorrent or manual blackhole like you gonna see in trace log)

### Steps To Reproduce

My script (his name is MySuperScript.sh):

```bash
#!/bin/bash
if [ ${sonarr_eventtype} = "Test" ]; then
  >&2 echo "[Debug] Testing script"
  exit
fi
echo "{\"file\": \"$sonarr_episodefile_path\",\"seriesid\": \"$sonarr_series_id\"}" >> /config/CustomScript/list
```

### Environment

```markdown
- OS: Almost Latest docker version (v4 linuxserver)
- Sonarr: V4 4.0.0.646
- Docker Install: Yes
- Using Reverse Proxy: Yes
- Browser: Firefox
```

### What branch are you running?

Develop

### Trace Logs?

2023-08-31 13:40:53.3|Debug|VideoFileInfoReader|Getting media info from /Blackhole/Sonarr/Ahsoka (2023) - S01E01 - Part One - Master and Apprentice WEBDL-1080p h264 EAC3[FR+EN] [FR+EN].mkv
2023-08-31 13:40:53.3|Debug|DetectSample|Runtime is over 90 seconds
2023-08-31 13:40:53.3|Trace|ConfigService|Using default config value for 'downloadclientworkingfolders' defaultValue:'_UNPACK_|_FAILED_'
2023-08-31 13:40:53.3|Debug|ImportDecisionMaker|File accepted
2023-08-31 13:40:53.3|Debug|EpisodeFileMovingService|Moving episode file: /Blackhole/Sonarr/Ahsoka (2023) - S01E01 - Part One - Master and Apprentice WEBDL-1080p h264 EAC3[FR+EN] [FR+EN].mkv to /Medias/Series/Ahsoka (2023)/Season 01/Ahsoka (2023) - S01E01 - Part One - Master and Apprentice WEBDL-1080p h264 EAC3[FR+EN] [FR+EN].mkv
2023-08-31 13:40:53.3|Trace|ConfigService|Using default config value for 'usescriptimport' defaultValue:'False'
2023-08-31 13:40:53.3|Debug|DiskTransferService|Move [/Blackhole/Sonarr/Ahsoka (2023) - S01E01 - Part One - Master and Apprentice WEBDL-1080p h264 EAC3[FR+EN] [FR+EN].mkv] > [/Medias/Series/Ahsoka (2023)/Season 01/Ahsoka (2023) - S01E01 - Part One - Master and Apprentice WEBDL-1080p h264 EAC3[FR+EN] [FR+EN].mkv]
2023-08-31 13:40:53.3|Trace|DiskProvider|Moved '/Blackhole/Sonarr/Ahsoka (2023) - S01E01 - Part One - Master and Apprentice WEBDL-1080p h264 EAC3[FR+EN] [FR+EN].mkv' -> '/Medias/Series/Ahsoka (2023)/Season 01/Ahsoka (2023) - S01E01 - Part One - Master and Apprentice WEBDL-1080p h264 EAC3[FR+EN] [FR+EN].mkv' using Syscall.rename
2023-08-31 13:40:53.3|Trace|ConfigService|Using default config value for 'filedate' defaultValue:'None'
2023-08-31 13:40:53.3|Trace|ConfigService|Using default config value for 'setpermissionslinux' defaultValue:'False'
2023-08-31 13:40:53.4|Trace|EventAggregator|Publishing EpisodeFileAddedEvent
2023-08-31 13:40:53.4|Trace|EventAggregator|EpisodeFileAddedEvent -> EpisodeService
2023-08-31 13:40:53.4|Trace|EventAggregator|Publishing ModelEvent<Episode>
2023-08-31 13:40:53.4|Trace|EventAggregator|ModelEvent<Episode> -> CutoffController
2023-08-31 13:40:53.4|Trace|EventAggregator|ModelEvent<Episode> <- CutoffController
2023-08-31 13:40:53.4|Trace|EventAggregator|ModelEvent<Episode> -> MissingController
2023-08-31 13:40:53.4|Trace|EventAggregator|ModelEvent<Episode> <- MissingController
2023-08-31 13:40:53.4|Trace|EventAggregator|ModelEvent<Episode> -> EpisodeController
2023-08-31 13:40:53.4|Trace|EventAggregator|ModelEvent<Episode> <- EpisodeController
2023-08-31 13:40:53.4|Trace|EventAggregator|ModelEvent<Episode> -> CalendarController
2023-08-31 13:40:53.4|Trace|EventAggregator|ModelEvent<Episode> <- CalendarController
2023-08-31 13:40:53.4|Debug|EpisodeService|Linking [Season 01/Ahsoka (2023) - S01E01 - Part One - Master and Apprentice WEBDL-1080p h264 EAC3[FR+EN] [FR+EN].mkv] > [[73416]Part One: Master and Apprentice]
2023-08-31 13:40:53.4|Trace|EventAggregator|EpisodeFileAddedEvent <- EpisodeService
2023-08-31 13:40:53.4|Trace|EventAggregator|EpisodeFileAddedEvent -> EpisodeFileController
2023-08-31 13:40:53.4|Trace|EventAggregator|EpisodeFileAddedEvent <- EpisodeFileController
2023-08-31 13:40:53.4|Trace|ConfigService|Using default config value for 'importextrafiles' defaultValue:'False'
2023-08-31 13:40:53.4|Trace|EventAggregator|Publishing EpisodeImportedEvent
2023-08-31 13:40:53.4|Trace|EventAggregator|EpisodeImportedEvent -> NotificationService
2023-08-31 13:40:53.4|Debug|NotificationService|No tags set for this notification.
2023-08-31 13:40:53.4|Trace|ConfigService|Using default config value for 'applicationurl' defaultValue:''
2023-08-31 13:40:53.4|Trace|EventAggregator|Publishing ProviderStatusChangedEvent<INotification>
2023-08-31 13:40:53.4|Warn|NotificationService|Unable to send OnDownload notification to: MySuperScript

[v4.0.0.646] NzbDrone.Core.Datastore.ModelNotFoundException: Tag with ID 1 does not exist
   at NzbDrone.Core.Datastore.BasicRepository`1.Get(Int32 id) in ./Sonarr.Core/Datastore/BasicRepository.cs:line 104
   at NzbDrone.Core.Notifications.CustomScript.CustomScript.<>c__DisplayClass14_0.<OnDownload>b__0(Int32 t) in ./Sonarr.Core/Notifications/CustomScript/CustomScript.cs:line 119
   at System.Linq.Enumerable.SelectEnumerableIterator`2.MoveNext()
   at System.String.Join(String separator, IEnumerable`1 values)
   at NzbDrone.Core.Notifications.CustomScript.CustomScript.OnDownload(DownloadMessage message) in ./Sonarr.Core/Notifications/CustomScript/CustomScript.cs:line 119
   at NzbDrone.Core.Notifications.NotificationService.Handle(EpisodeImportedEvent message) in ./Sonarr.Core/Notifications/NotificationService.cs:line 179

2023-08-31 13:40:53.4|Debug|NotificationService|No tags set for this notification.
2023-08-31 13:40:53.4|Trace|ConfigService|Using default config value for 'applicationurl' defaultValue:''
2023-08-31 13:40:53.4|Trace|HttpClient|Req: [POST] http://autoscan:3030/triggers/sonarr
2023-08-31 13:40:53.4|Trace|ConfigService|Using default config value for 'proxyenabled' defaultValue:'False'
2023-08-31 13:40:53.4|Trace|EventAggregator|Publishing HealthCheckCompleteEvent
2023-08-31 13:40:53.4|Trace|EventAggregator|HealthCheckCompleteEvent -> HealthController
2023-08-31 13:40:53.4|Trace|EventAggregator|HealthCheckCompleteEvent <- HealthController
2023-08-31 13:40:53.4|Trace|EventAggregator|HealthCheckCompleteEvent ~> NotificationService
2023-08-31 13:40:53.4|Trace|EventAggregator|HealthCheckCompleteEvent <~ NotificationService
2023-08-31 13:40:53.4|Trace|HttpClient|Res: HTTP/1.1 [POST] http://autoscan:3030/triggers/sonarr: 200.OK (0 bytes) (28 ms)
2023-08-31 13:40:53.4|Trace|EventAggregator|EpisodeImportedEvent <- NotificationService
2023-08-31 13:40:53.4|Trace|EventAggregator|EpisodeImportedEvent -> HistoryService
2023-08-31 13:40:53.5|Trace|EventAggregator|EpisodeImportedEvent <- HistoryService
2023-08-31 13:40:53.5|Trace|EventAggregator|EpisodeImportedEvent -> DownloadMonitoringService
2023-08-31 13:40:53.5|Trace|EventAggregator|EpisodeImportedEvent <- DownloadMonitoringService
2023-08-31 13:40:53.5|Trace|EventAggregator|EpisodeImportedEvent -> DownloadHistoryService
2023-08-31 13:40:53.5|Trace|EventAggregator|EpisodeImportedEvent <- DownloadHistoryService
2023-08-31 13:40:53.5|Trace|EventAggregator|EpisodeImportedEvent -> CutoffController
2023-08-31 13:40:53.5|Trace|EventAggregator|EpisodeImportedEvent <- CutoffController
2023-08-31 13:40:53.5|Trace|EventAggregator|EpisodeImportedEvent -> MissingController
2023-08-31 13:40:53.5|Trace|EventAggregator|EpisodeImportedEvent <- MissingController
2023-08-31 13:40:53.5|Trace|EventAggregator|EpisodeImportedEvent -> SeriesController
2023-08-31 13:40:53.5|Trace|EventAggregator|EpisodeImportedEvent <- SeriesController
2023-08-31 13:40:53.5|Trace|EventAggregator|EpisodeImportedEvent -> EpisodeController
2023-08-31 13:40:53.5|Trace|Http|Req: 3824 [GET] /api/v3/health (from 172.*.*.164 Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Safari/537.36)
2023-08-31 13:40:53.5|Trace|Http|Res: 3824 [GET] /api/v3/health: 200.OK (1 ms)
2023-08-31 13:40:53.5|Trace|EventAggregator|EpisodeImportedEvent <- EpisodeController
2023-08-31 13:40:53.5|Trace|EventAggregator|EpisodeImportedEvent -> CalendarController
2023-08-31 13:40:53.5|Debug|Api|[GET] /api/v3/health: 200.OK (1 ms)
2023-08-31 13:40:53.5|Trace|EventAggregator|EpisodeImportedEvent <- CalendarController
2023-08-31 13:40:53.5|Trace|EventAggregator|Publishing HealthCheckCompleteEvent
2023-08-31 13:40:53.5|Trace|EventAggregator|HealthCheckCompleteEvent -> HealthController
2023-08-31 13:40:53.5|Debug|CompletedDownloadService|All episodes were imported for Ahsoka (2023) - S01E01 - Part One - Master and Apprentice WEBDL-1080p h264 EAC3[FR+EN] [FR+EN].mkv
2023-08-31 13:40:53.5|Trace|EventAggregator|HealthCheckCompleteEvent <- HealthController
2023-08-31 13:40:53.5|Trace|EventAggregator|Publishing DownloadCompletedEvent
2023-08-31 13:40:53.5|Trace|EventAggregator|DownloadCompletedEvent -> DownloadEventHub
2023-08-31 13:40:53.5|Trace|EventAggregator|HealthCheckCompleteEvent ~> NotificationService
2023-08-31 13:40:53.5|Debug|DownloadEventHub|[Ahsoka (2023) - S01E01 - Part One - Master and Apprentice WEBDL-1080p h264 EAC3[FR+EN] [FR+EN].mkv] Marking download as imported from blackhole
2023-08-31 13:40:53.5|Debug|DownloadEventHub|Torrent Blackhole does not support marking items as imported
2023-08-31 13:40:53.5|Debug|DownloadEventHub|[Ahsoka (2023) - S01E01 - Part One - Master and Apprentice WEBDL-1080p h264 EAC3[FR+EN] [FR+EN].mkv] Removing download from blackhole history
2023-08-31 13:40:53.5|Trace|TorrentBlackhole|[Ahsoka (2023) - S01E01 - Part One - Master and Apprentice WEBDL-1080p h264 EAC3[FR+EN] [FR+EN].mkv] File or folder '/Blackhole/Sonarr/Ahsoka (2023) - S01E01 - Part One - Master and Apprentice WEBDL-1080p h264 EAC3[FR+EN] [FR+EN].mkv' doesn't exist, skipping cleanup.
2023-08-31 13:40:53.5|Trace|EventAggregator|DownloadCompletedEvent <- DownloadEventHub
2023-08-31 13:40:53.5|Trace|EventAggregator|DownloadCompletedEvent -> DownloadHistoryService
2023-08-31 13:40:53.5|Trace|EventAggregator|HealthCheckCompleteEvent <~ NotificationService
2023-08-31 13:40:53.5|Trace|EventAggregator|DownloadCompletedEvent <- DownloadHistoryService
2023-08-31 13:40:53.5|Trace|EventAggregator|Publishing DownloadsProcessedEvent
2023-08-31 13:40:53.5|Trace|EventAggregator|DownloadsProcessedEvent -> DownloadMonitoringService
2023-08-31 13:40:53.5|Trace|EventAggregator|Publishing TrackedDownloadRefreshedEvent
2023-08-31 13:40:53.5|Trace|EventAggregator|TrackedDownloadRefreshedEvent -> QueueService
2023-08-31 13:40:53.5|Trace|EventAggregator|Publishing QueueUpdatedEvent
2023-08-31 13:40:53.5|Trace|EventAggregator|QueueUpdatedEvent -> QueueController
2023-08-31 13:40:53.5|Trace|EventAggregator|QueueUpdatedEvent <- QueueController
2023-08-31 13:40:53.5|Trace|EventAggregator|QueueUpdatedEvent -> QueueDetailsController
2023-08-31 13:40:53.5|Trace|EventAggregator|QueueUpdatedEvent <- QueueDetailsController
2023-08-31 13:40:53.5|Trace|EventAggregator|QueueUpdatedEvent -> QueueStatusController
2023-08-31 13:40:53.5|Trace|EventAggregator|QueueUpdatedEvent <- QueueStatusController
2023-08-31 13:40:53.5|Trace|EventAggregator|TrackedDownloadRefreshedEvent <- QueueService
2023-08-31 13:40:53.5|Trace|EventAggregator|DownloadsProcessedEvent <- DownloadMonitoringService
2023-08-31 13:40:53.5|Trace|CommandQueueManager|Updating command status
2023-08-31 13:40:53.5|Trace|EventAggregator|DownloadsProcessedEvent ~> NotificationService
2023-08-31 13:40:53.5|Trace|EventAggregator|DownloadsProcessedEvent <~ NotificationService
2023-08-31 13:40:53.5|Trace|EventAggregator|Publishing CommandExecutedEvent
2023-08-31 13:40:53.5|Trace|EventAggregator|CommandExecutedEvent -> TaskManager
2023-08-31 13:40:53.5|Trace|EventAggregator|CommandExecutedEvent <- TaskManager
2023-08-31 13:40:53.5|Trace|EventAggregator|CommandExecutedEvent -> TaskController
2023-08-31 13:40:53.5|Trace|EventAggregator|CommandExecutedEvent <- TaskController
2023-08-31 13:40:53.5|Trace|EventAggregator|CommandExecutedEvent -> QualityDefinitionController
2023-08-31 13:40:53.5|Trace|EventAggregator|CommandExecutedEvent <- QualityDefinitionController
2023-08-31 13:40:53.5|Trace|CommandExecutor|ProcessMonitoredDownloadsCommand <- DownloadProcessingService [00:00:00.6473000]
2023-08-31 13:40:53.5|Trace|Http|Req: 3825 [GET] /api/v3/health (from 172.*.*.102 Mozilla/5.0 (Windows NT 10.0; rv:109.0) Gecko/20100101 Firefox/116.0)
2023-08-31 13:40:53.5|Trace|Http|Res: 3825 [GET] /api/v3/health: 200.OK (0 ms)
2023-08-31 13:40:53.5|Debug|Api|[GET] /api/v3/health: 200.OK (0 ms)

### Anything else?

I just update now to the latest 4.0.0.651 and gonna see if the problem is here Wednesday when the new episode gonna be release.

**Comment by markus101** [comment]
Please use one of the support channels: [forums](https://forums.sonarr.tv/), [subreddit](https://www.reddit.com/r/sonarr/), [discord ](https://discord.sonarr.tv/), or [IRC](https://web.libera.chat/?channels=#sonarr) for support/questions.

> 13:40:53.4|Warn|NotificationService|Unable to send OnDownload notification to: MySuperScript

> [v4.0.0.646] NzbDrone.Core.Datastore.ModelNotFoundException: Tag with ID 1 does not exist

You have a bad/m tag in there, running the housekeeping task should clean it up, but you may need to remove and readd the series.
